### PR TITLE
fix(d.ts): add Bone toJSON & toObject API

### PR DESCRIPTION
**Pull request by nightink**

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -543,6 +543,28 @@ export class Bone {
...
+  /**
+   * This is the loyal twin of {@link Bone#toJSON} because when generating the result object, the raw values of attributes are used, instead of the values returned by custom getters (if any).
+   * {@link Bone#toObject} might be called on descents of Bone that does not have attributes defined on them directly, hence for..in is preferred.
+   * - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties
+   * @example
+   * const post = await Post.first
+   * post.toObject()  // => { id: 1, ... }
+   * @return {Object}
+   */
+  toObject(): Record<string, any>;
```
Comment (cyjake) [inline types/index.d.ts:567]: `Record<string, Literal | JSON>`

**Review by cyjake — COMMENTED** [review]
Could you add some test code in types/leoric-tests.ts as well?

File: types/index.d.ts (hunk 1 above)
Comment (nightink) [reply types/index.d.ts]: Literal includes JSON

**Comment by nightink** [comment]
done

File: types/leoric-tests.ts [hunk 2]
```diff
@@ -11,7 +11,9 @@ class User extends Bone {
 }
 
 async function main() {
-  await User.create({ name: 'Stranger' })
+  const userBone = await User.create({ name: 'Stranger' });
+  console.log(userBone.toJSON(), userBone.toObject());
```
Comment (JimmyDaddy) [inline types/leoric-tests.ts:15]: `console.log` should be removed

Comment (nightink) [reply types/leoric-tests.ts]: done

**Comment by nightink** [comment]
Also added typescript type definition unit test

File: src/hint.js [hunk 3]
```diff
@@ -200,9 +200,9 @@ class IndexHint {
 
     const result = [];
     for (const key in grouped) {
-      const indices = grouped[key].reduce((result, index) => {
-        if (!result.includes(index)) result.push(index);
-        return result;
+      const indices = grouped[key].reduce((arr, index) => {
+        if (!arr.includes(index)) arr.push(index);
+        return arr;
```
Comment (nightink) [inline src/hint.js:205]: fix `npm run elint` error

**Comment by cyjake** [comment]
> Also added typescript type definition unit test

**Pull request merged**